src/raftstore/store/peer_storage.rs

```diff
@@ -319,25 +319,31 @@ impl PeerStorage {
         let mut next_index = low;
         let mut exceeded_max_size = false;
 
-        let start_key = keys::raft_log_key(self.get_region_id(), low);
-
-        if low + 1 == high {
+        if low + 8 >= high {
             // If election happens in inactive regions, they will just try
             // to fetch one empty log.
             let handle = self.engine.cf_handle(CF_RAFT).unwrap();
-            match box_try!(self.engine.get_cf(handle, &start_key)) {
-                None => return Err(RaftError::Store(StorageError::Unavailable)),
-                Some(v) => {
-                    let mut entry = Entry::new();
-                    box_try!(entry.merge_from_bytes(&v));
-                    assert_eq!(entry.get_index(), low);
-                    return Ok(vec![entry]);
+            for i in low..high {
+                let key = keys::raft_log_key(self.get_region_id(), i);
+                match box_try!(self.engine.get_cf(handle, &key)) {
```

Comment:
I think maybe we can try tailing iterator here. 

https://github.com/facebook/rocksdb/wiki/Tailing-Iterator

@zhangjinpeng1987